x-pack/plugins/ml/public/file_datavisualizer/components/edit_flyout/overrides_validation.js

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { i18n } from '@kbn/i18n';
+
+const INDETERMINATE_FIELD_PLACEHOLDER = '?';
+const FRACTIONAL_SECOND_SEPARATORS = ':.,';
+
+const VALID_LETTER_GROUPS = {
+  'yyyy': true,
+  'yy': true,
+  'M': true,
+  'MM': true,
+  // The simple regex here is based on the fact that the %{MONTH} Grok pattern only matches English and German month names
+  'MMM': true,
+  'MMMM': true,
+  'd': true,
+  'dd': true,
+  // The simple regex here is based on the fact that the %{DAY} Grok pattern only matches English and German day names
+  'EEE': true,
+  'EEEE': true,
+  'H': true,
+  'HH': true,
+  'h': true,
+  'mm': true,
+  'ss': true,
+  'a': true,
+  'XX': true,
+  'XXX': true,
+  'zzz': true,
+};
+// TODO: ensure this is the best way to do this check
+function isLetter(str) {
+  return str.length === 1 && str.match(/[a-z]/i);
+}
+
+export function isTimestampFormatValid(timestampFormat) {
+  const result = { isValid: true, errorMessage: null };
+
+  if (timestampFormat.indexOf('\n') >= 0 || timestampFormat.indexOf('\r') >= 0) {
```

Comment:
Maybe this part of the validation (lines 43-52) could be removed and instead set a property on the text box that prevents it containing multiple lines?  (Is it even possible to stop newlines being added into the text box?  Note my complete absence of UI development skills here.  If there's no way to ban newlines from the text box then leave the check here.)

Reply:
Good call - I was in the process of removing this as I found it wasn't possible to add multiple lines to the EUI component I'm using so it shouldn't be a problem. 👍

Reply:
Removed in 2adb1002d6ef80fb467803db00d1273545636951